runtime-sdk/modules/rofl-market/src/lib.rs

```diff
@@ -331,15 +331,18 @@ impl<Cfg: Config> Module<Cfg> {
     }
 
     #[handler(call = "roflmarket.InstanceCreate")]
-    fn tx_instance_create<C: Context>(ctx: &C, body: types::InstanceCreate) -> Result<(), Error> {
+    fn tx_instance_create<C: Context>(
+        ctx: &C,
+        body: types::InstanceCreate,
+    ) -> Result<types::InstanceId, Error> {
         <C::Runtime as Runtime>::Core::use_tx_gas(Cfg::GAS_COST_CALL_INSTANCE_CREATE)?;
 
         if body.term_count == 0 {
             return Err(Error::InvalidArgument);
         }
 
         if !ctx.should_execute_contracts() {
-            return Ok(());
+            return Ok(Default::default());
```

Comment:
It is hard for the caller to know if returned value was received from the happy path or from this branch. Returning error or option seems better.

Reply:
This only happens during internal simulation or transaction checks. I agree that it is somewhat awkward, but in these cases this value is _always_ discarded (as it originates from the gas estimator). So the value doesn't matter but it must satisfy the type system.

Changing it to an option doesn't make sense as in any externally visible call, this will never be `None`. We could probably use special errors for these cases, but currently this could maybe interfere with simulation and therefore it should be handled in a separate issue. Other places also handle it similarly.